liota/dccs/iotcc.py

```diff
@@ -177,9 +177,8 @@ def create_relationship(self, reg_entity_parent, reg_entity_child):
             Parameters:
             - obj: The object that has just obtained an UUID
         """
-        # sanity check: must be RegisteredEntity or RegisteredMetricRegisteredMetric
-        if (not isinstance(reg_entity_parent, RegisteredEntity) \
-                    and not isinstance(reg_entity_parent, RegisteredMetric)) \
```

Comment:
What is the reason for removing the metric check or is it by mistakenly removed?

Reply:
reg_entity_parent can't be a RegisteredMetric.

Reply:
The check should exist for metric as well because the inputs are passed from the user code and we don't have any control on it.

Reply:
We can not allow a user to have a RegisteredMetric as a parent for any other RegisteredMetric or RegisteredEntity. Hence, this check is removed so that even if a user passes RegisteredMetric as a parent, we can throw an error. child object can still be RegisteredMetric or RegisteredEntity, and that check is continued.

Reply:
Got it the extra brackets confused me in the CL, it should directly be;

not isinstance(reg_entity_parent, RegisteredEntity)

Reply:
Yes, you are right. The extra bracket just helps readability, that the not is applicable to the first condition only. Technically, it can be removed, but will keep it for now.